### PR TITLE
Add discount_allocations field to order's line_item

### PR DESCRIPTION
**Pull request by milos-matijasevic**
Adding discount_allocations in LineItem struct, regarding newest [Shopify api](https://shopify.dev/docs/admin-api/rest/reference/orders/order).

### CODE REVIEW

File: order.go [hunk 1]
```diff
@@ -171,32 +171,49 @@ type DiscountCode struct {
...
+	Properties                 []NoteAttribute       `json:"properties,omitempty"`
+	ProductExists              bool                  `json:"product_exists,omitempty"`
+	FulfillableQuantity        int                   `json:"fulfillable_quantity,omitempty"`
+	Grams                      int                   `json:"grams,omitempty"`
+	FulfillmentStatus          string                `json:"fulfillment_status,omitempty"`
+	TaxLines                   []TaxLine             `json:"tax_lines,omitempty"`
+	OriginLocation             *Address              `json:"origin_location,omitempty"`
+	DestinationLocation        *Address              `json:"destination_location,omitempty"`
+	AppliedDiscount            *AppliedDiscount      `json:"applied_discount,omitempty"`
+	DiscoundAllocations        []DiscoundAllocations `json:"discount_allocations,omitempty"`
```
Comment (sandersjessica) [inline order.go:200]: There are a few spelling mistakes on line 200 and 203. Can you please replace `Discound` with `Discount`?

**Review by sandersjessica — CHANGES_REQUESTED** [review]

File: order.go (hunk 1 above)
Comment (milos-matijasevic) [reply order.go]: thanks

**Comment by milos-matijasevic** [comment]
Should i add a test for this?

**Review by ysurtayev-bold — APPROVED** [review]
Now looks good from my prospective. 
@sandersjessica what do you think?

**Review by sandersjessica — APPROVED** [review]

**Comment by gordcurrie** [comment]
Thanks for the contribution, there are a few PRs queued up we'll look at doing a release when they're ready to go.

**Pull request merged**